markermodel.h

```diff
@@ -0,0 +1,324 @@
+    const std::string worldID = "world";
+    const std::string orangeHouseID = "orangeHouse";
+    const std::string adaHouseID = "adaHouse";
+
+    const std::string world2camID = "world2cam";
+    const std::string world2orangeHouseID = "world2orangeHouse";
+    const std::string world2adaHouseID = "world2adaHouse";
+
+    // Thread which runs the monitoring.
+    QThread monitorThread;
+
+signals:
+
+    // Signals for the monitor.
+
+    // Object emits this signal everytime it received an update for a tranformation through
+    // updateLinkNow(..) from the QML model to inform the monitor also about the update.
+    void linkUpdate(const std::string &srcFrame, const std::string &dstFrame, const Timestamp &ts,
+                    const qPair &transf, const float &conf);
+
+    // Object can emit this signal whenever a calculated transformation (result, output to QML model) was updated.
+    void transformationUpdate(const std::string &transID, const Timestamp &ts, const QMatrix4x4 &trans,
+                              const float &avgLinkConfidence, const float &avgDistanceToEntry);
+
+    void registerLinkUpdateToMonitor(const std::string &srcFrame, const std::string &destFrame);
+    void registerTransformationToMonitor(const std::string &transID);
+
+    void startMonitor();
+    void stopMonitor();
+
+    // Signals for QML model
+    void transformationsUpdated();
+};
+
+/*****************************
+ * LINK UPDATE               *
+ *****************************/
+
+// Stores a single update of a link direct from the tracker.
+struct LinkUpdate {
+    Timestamp time;
+    qPair transformation;
+    double confidence;
+};
+
+/*****************************
+ * TRANSFORMATION UPDATE     *
+ *****************************/
+
+// Stores a single update of a transformation used to draw an object.
+struct TransformationUpdate  {
+    Timestamp time;
+    qPair transformation;
+    float avgLinkConfidence;
+    float maxDistanceToEntry;
+};
+
+/*******************************
+ * ANALYSIS SINGLE RESULT      *
+ ******************************/
+
+// Stores a single result of an analysis.
+struct AnalysisSingleResult {
+
+    long tms;               // Elapsed time since analysis started in ms.
+
+    qPair tf;                       // Transformation
+
+    /* Values yield from the comparision of two transformation. For further
+     * information check the function compareqPair(..) and the implementation of the corrensponding
+     * analyses using this datatype to store their results. */
+    double ratioLenT;
+    double distanceNormalT;
+    double distanceNormalPointR;
+    double distanceNormalR;
+
+    double conf;                    // Confidence
+    double distanceToEntry;
+};
+
+/*****************************
+ * TRANSFORMATION UPDATE     *
+ *****************************/
+
+// Basic type of an analysis.
+struct Analysis{
+
+    // Don't allow creation of an object of this type.
+    virtual ~Analysis(){}
```

Comment:
Do not write an empty destructor like this, just tell the compiler to use the default one: `virtual ~Analysis() = default;`.